projects/plugins/jetpack/class.jetpack.php

```diff
@@ -3467,11 +3470,18 @@ public static function disconnect( $update_activated_state = true ) {
 		// let's make sure to not disconnect the production site.
 		$connection->disconnect_site( ! Identity_Crisis::validate_sync_error_idc_option() );
 
-		Identity_Crisis::clear_all_idc_options();
-
 		if ( $update_activated_state ) {
 			Jetpack_Options::update_option( 'activated', 4 );
 		}
+	}
+
+	/**
+	 * Happens after a successfull disconnection.
+	 *
+	 * @static
+	 */
+	public static function jetpack_site_disconnected() {
```

Comment:
@mdbitz Ideally we don't even need this here. Both `identity-crisis` and `sync` packages could make use of the `jetpack_site_disconnected` action and handle this internally.

However I was a bit reluctant on making too many changes without discussing it first.

Reply:
I'd have changes to IDC and Sync related logic as a separate PR.